### DISCUSSION TITLE
Prompt caching

### DISCUSSION THREAD

**Discussion by ebudmada** [MadcowD/ell#200] [Q&A]
Can we do prompt caching with ell ?

**Comment by gwpl** [comment]
> Can we do prompt caching with ell ?

To not be confused, some providers offer prompt caching on server side. I guess that's not what you think?

If you want to cache function calls in your script, then it's library independent I guess?

Have you gave a shot to classical Memoization approaches if they work for you:
https://stackoverflow.com/questions/1988804/what-is-memoization-and-how-can-i-use-it-in-python
like [`functools.cache`](https://docs.python.org/3/library/functools.html#functools.cache) , DIY `@memoize` decorators,

**Comment by gwpl** [comment]
cross-linking: https://github.com/MadcowD/ell/issues/389
